frontend/workflows/dynamodb/src/update-capacity.tsx

```diff
@@ -0,0 +1,254 @@
+                        .min(Number(gsi.provisionedThroughput.readCapacityUnits)),
+                    },
+                  },
+                  {
+                    name: "write",
+                    value: gsi.provisionedThroughput.writeCapacityUnits,
+                    input: {
+                      type: "number",
+                      key: `write_capacity_units,${gsi.name}`,
+                      validation: number()
+                        .integer()
+                        .transform(value => (Number.isNaN(value) ? 0 : value))
+                        .min(Number(gsi.provisionedThroughput.writeCapacityUnits)),
+                    },
+                  },
+                ]}
+              />
+            </TableRow>
+          ))}
+        </Table>
+      </Container>
+
+      {/* TO DO: conditionally render the override checkbox depending on workflow config prop */}
+      <CheckboxPanel
+        header="To override the safety limits for scaling, check the box below."
+        onChange={state => capacityUpdates.updateData("ignore_maximums", state)}
+        options={{
+          "Override maximum limits": false,
+        }}
+      />
+
+      <ButtonGroup>
+        <Button text="Back" variant="neutral" onClick={() => onBack()} />
+        <Button text="Update" onClick={onSubmit} />
+      </ButtonGroup>
+    </WizardStep>
+  );
+};
+
+const Confirm: React.FC<WizardChild> = () => {
+  const updateCapacityOutput = useDataLayout("updateCapacityOutput");
+  const scalingResults = updateCapacityOutput.displayValue()?.data?.table;
+  let statusList = [];
+  if (!_.isEmpty(scalingResults)) {
+    statusList.push(scalingResults);
+    statusList = statusList.concat(scalingResults.globalSecondaryIndexes);
+  }
+
+  return (
+    <WizardStep error={updateCapacityOutput.error} isLoading={updateCapacityOutput.isLoading}>
+      <Table columns={["Name", "Status"]}>
+        {statusList.map(s => (
+          <TableRow key={s.name}>
+            {s.name}
+            {s.status}
+          </TableRow>
+        ))}
+      </Table>
+    </WizardStep>
+  );
+};
+
+const UpdateCapacity: React.FC<WorkflowProps> = ({ resolverType }) => {
+  const dataLayout = {
+    resourceData: {},
+    capacityUpdates: {},
+    updateCapacityOutput: {
+      deps: ["resourceData", "capacityUpdates"],
+      hydrator: (resourceData, capacityUpdates) => {
+        const tableArgs = {
+          table_name: resourceData.name,
+          region: resourceData.region,
+          ignore_maximums:
+            "ignore_maximums" in capacityUpdates ? capacityUpdates.ignore_maximums : false,
+        };
+
+        let changeArgs: {};
+        if (capacityUpdates.table_throughput) {
+          changeArgs = {
+            ...changeArgs,
+            table_throughput: {
+              read_capacity_units: capacityUpdates.table_throughput.read
+                ? capacityUpdates.table_throughput.read
+                : resourceData.provisionedThroughput.readCapacityUnits,
```

Comment:
nit simplify with null coalesce
```suggestion
              read_capacity_units: capacityUpdates.table_throughput?.read ?? resourceData.provisionedThroughput.readCapacityUnits,
```